internal/observability/graph_test.go

```diff
@@ -47,15 +48,15 @@ func TestEdgeNopFallbacks(t *testing.T) {
 	}
 
 	// Should succeed, covered by middleware tests.
-	_ = newEdge(zap.NewNop(), reg, req, _directionOutbound)
+	_ = newEdge(zap.NewNop(), meter, req, _directionOutbound)
 
 	// Should fall back to no-op metrics.
-	e := newEdge(zap.NewNop(), reg, req, _directionOutbound)
-	assert.NotNil(t, e.calls, "Expected to fall back to no-op metrics.")
-	assert.NotNil(t, e.successes, "Expected to fall back to no-op metrics.")
-	assert.NotNil(t, e.callerFailures, "Expected to fall back to no-op metrics.")
-	assert.NotNil(t, e.serverFailures, "Expected to fall back to no-op metrics.")
-	assert.NotNil(t, e.latencies, "Expected to fall back to no-op metrics.")
-	assert.NotNil(t, e.callerErrLatencies, "Expected to fall back to no-op metrics.")
-	assert.NotNil(t, e.serverErrLatencies, "Expected to fall back to no-op metrics.")
+	e := newEdge(zap.NewNop(), meter, req, _directionOutbound)
+	assert.Nil(t, e.calls, "Expected to fall back to no-op metrics.")
+	assert.Nil(t, e.successes, "Expected to fall back to no-op metrics.")
+	assert.Nil(t, e.callerFailures, "Expected to fall back to no-op metrics.")
+	assert.Nil(t, e.serverFailures, "Expected to fall back to no-op metrics.")
+	assert.Nil(t, e.latencies, "Expected to fall back to no-op metrics.")
+	assert.Nil(t, e.callerErrLatencies, "Expected to fall back to no-op metrics.")
+	assert.Nil(t, e.serverErrLatencies, "Expected to fall back to no-op metrics.")
```

Comment:
Is this expected? This is the exact opposite of the previous behavior.

Reply:
Nil metrics are valid now, before there was an explicit nil implementation.

Reply:
Instead of asserting that these pointers are nil, it may be better to assert that using each counter/vector/histogram doesn't panic.